server/blob/blob-storage-strategy/src/main/java/org/apache/james/server/blob/deduplication/BlobGCService.java

```diff
@@ -0,0 +1,109 @@
+import org.apache.james.mailbox.cassandra.mail.AttachmentBlobReferenceSource;
+import org.apache.james.mailbox.cassandra.mail.MessageBlobReferenceSource;
+import org.apache.james.mailrepository.cassandra.MailRepositoryBlobReferenceSource;
+import org.apache.james.queue.rabbitmq.view.cassandra.MailQueueViewBlobReferenceSource;
+import org.apache.james.server.blob.deduplication.BloomFilterGCAlgorithm.Context;
+import org.apache.james.task.Task;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class BlobGCService {
+
+    public static final int EXPECTED_BLOB_COUNT = 1_000_000;
+    public static final double ASSOCIATED_PROBABILITY = 0.8;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BlobGCService.class);
+
+    private final BlobStoreDAO blobStoreDAO;
+    private final GenerationAwareBlobId.Factory generationAwareBlobIdFactory;
+    private final GenerationAwareBlobId.Configuration generationAwareBlobIdConfiguration;
+
+    private final MessageBlobReferenceSource messageBlobReferenceSource;
+    private final MailRepositoryBlobReferenceSource mailRepositoryBlobReferenceSource;
+    private final MailQueueViewBlobReferenceSource mailQueueViewBlobReferenceSource;
+    private final AttachmentBlobReferenceSource attachmentBlobReferenceSource;
+    private final Clock clock;
+
+    @Inject
+    public BlobGCService(BlobStoreDAO blobStoreDAO,
+                         GenerationAwareBlobId.Factory generationAwareBlobIdFactory,
+                         GenerationAwareBlobId.Configuration generationAwareBlobIdConfiguration,
+                         MessageBlobReferenceSource messageBlobReferenceSource,
+                         MailRepositoryBlobReferenceSource mailRepositoryBlobReferenceSource,
+                         MailQueueViewBlobReferenceSource mailQueueViewBlobReferenceSource,
+                         AttachmentBlobReferenceSource attachmentBlobReferenceSource,
+                         Clock clock) {
+
+        this.blobStoreDAO = blobStoreDAO;
+        this.generationAwareBlobIdFactory = generationAwareBlobIdFactory;
+        this.generationAwareBlobIdConfiguration = generationAwareBlobIdConfiguration;
+        this.messageBlobReferenceSource = messageBlobReferenceSource;
+        this.mailRepositoryBlobReferenceSource = mailRepositoryBlobReferenceSource;
+        this.mailQueueViewBlobReferenceSource = mailQueueViewBlobReferenceSource;
+        this.attachmentBlobReferenceSource = attachmentBlobReferenceSource;
+        this.clock = clock;
+    }
+
+    public Mono<Result> gc() {
+        BloomFilterGCAlgorithm gcAlgorithm = new BloomFilterGCAlgorithm(
+            getBlobReferenceSource(),
+            blobStoreDAO,
+            generationAwareBlobIdFactory,
+            generationAwareBlobIdConfiguration,
+            clock);
+
+        Context context = new Context(EXPECTED_BLOB_COUNT, ASSOCIATED_PROBABILITY);
```

Comment:
TODO EXPECTED_BLOB_COUNT and ASSOCIATED_PROBABILITY should not be hardcoded...

Reply:
Should we read it from `blob.properties` ?

Reply:
No, that should be parameters at task creation